test/performance/ThroughputPublisher.cpp

```diff
@@ -543,10 +558,19 @@ bool ThroughputPublisher::test(
                 mp_datapub->write((void*)throughput);
             }
         }
+        // Get end time
         t_end_ = std::chrono::steady_clock::now();
+        // Add the number of sent samples
         samples += demand;
-        std::this_thread::sleep_for(std::chrono::milliseconds(recovery_time_ms));
-        timewait_us += t_overhead_;
+
+        // If the batch took less than the recovery time, sleep for the difference recovery_time - batch_duration.
+        // Else, go ahead with next batch without time to recover.
+        if ((t_end_ - batch_start) <= std::chrono::milliseconds(recovery_time_ms))
+        {
+            std::this_thread::sleep_for(std::chrono::milliseconds(recovery_time_ms) - (t_end_ - batch_start));
+        }
```

Comment:
I would calculate the time to sleep first, and then do the sleep only if that time is greater than 0.